sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala

```diff
@@ -2790,7 +2790,9 @@ class ColumnExpressionSuite extends QueryTest with SharedSparkSession {
           ).foreach { case (end, start) =>
             val df = Seq((end, start)).toDF("end", "start")
             val daysBetween = Duration.ofDays(ChronoUnit.DAYS.between(start, end))
-            checkAnswer(df.select($"end" - $"start"), Row(daysBetween))
+            val r = df.select($"end" - $"start").toDF("diff")
+            checkAnswer(r, Row(daysBetween))
+            assert(r.schema === new StructType().add("diff", DayTimeIntervalType(0, 0)))
```

Comment:
Please, replace `DayTimeIntervalType(0, 0)` by `DayTimeIntervalType(DAY)`

Reply:
@Peng-Lei This change seems to include the change not only for `Date - Date` but also `Date + Interval` and `Interval + Date`. If you'd like to include them, could you add tests for them too?

Reply:
Let's exclude the `Date +/- Interval` changes from this PR.

Reply:
> could you add tests for them too

@sarutak ,Sorry I am wrong. First I should exclude the `Date +/- Interval` from this PR. Second Even if I wanted to change `Date +/- Interval`, I should convert the interval to the number of days `ExtractANSIIntervalDays)` and then use `DateAdd` expression as cloud-fan said.